helpers/controllers/submissionController.ts

```diff
@@ -22,34 +22,39 @@ export const createSubmission = async (
     const { challengeId, cliToken, diff, lessonId } = args
     const { id } = decode(cliToken)
     const submissionData = { diff, status: SubmissionStatus.Open }
-    const { lesson, challenge, user, ...submission } =
-      await prisma.submission.upsert({
-        where: {
-          userId_lessonId_challengeId: {
-            userId: Number(id),
-            lessonId,
-            challengeId
-          }
-        },
-        create: {
-          ...submissionData,
-          challengeId,
-          lessonId,
-          userId: Number(id)
+    const previousSubmission = await prisma.submission.findFirst({
+      where: {
+        challengeId,
+        lessonId,
+        user: {
+          id
         },
-        update: {
-          ...submissionData,
-          comments: {
-            deleteMany: {}
-          }
+        status: SubmissionStatus.Open
+      }
+    })
+    if (previousSubmission) {
+      await prisma.submission.update({
+        where: {
+          id: previousSubmission.id
         },
-        include: {
-          user: true,
-          lesson: true,
-          challenge: true
+        data: {
+          status: SubmissionStatus.Overwritten
         }
       })
-
+    }
+    const submission = await prisma.submission.create({
+      data: {
+        ...submissionData,
```

Comment:
If you want you can inline these values here. I only created this `submissionData` to avoid duplicated values in the upsert create and update.